Framework/Geometry/src/Math/Quadrilateral.cpp

```diff
@@ -11,10 +13,11 @@ using Kernel::V2D;
 //----------------------------------------------------------------------------------------------
 /** Constructor
  */
-Quadrilateral::Quadrilateral(const V2D &lowerLeft, const V2D &lowerRight,
-                             const V2D &upperRight, const V2D &upperLeft)
-    : ConvexPolygon(), m_lowerLeft(lowerLeft), m_lowerRight(lowerRight),
-      m_upperRight(upperRight), m_upperLeft(upperLeft) {}
+Quadrilateral::Quadrilateral(V2D lowerLeft, V2D lowerRight, V2D upperRight,
+                             V2D upperLeft)
+    : ConvexPolygon(), m_lowerLeft(std::move(lowerLeft)),
+      m_lowerRight(std::move(lowerRight)), m_upperRight(std::move(upperRight)),
+      m_upperLeft(std::move(upperLeft)) {}
```

Comment:
I removed `modernize-pass-by-value` from the list of tests in the [nightly build](http://builds.mantidproject.org/view/Static%20Analysis/job/clang-tidy_fixed/). While I still think it is an interesting optimization to, under the right circumstances, eliminate a copy, it may be best done on a case-by-case basis and justified with benchmarks.